backend/cfg_selectgen.ml

```diff
@@ -1414,8 +1328,14 @@ module Make (Target : Cfg_selectgen_target_intf.S) = struct
                     [||] [||])
               ids_and_rs)
       in
+      (* CR vlaviron: I hit an issue in the non-tail case where this
```

Comment:
Was there a reason this wasn't fixed at this point?  Can we add a test case?

Reply:
I didn't fix it because I didn't know whether I had wrongly mixed `add_empty_block_at_start` and `mark_as_trap_handler`, or if that was supposed to work but didn't for other reasons.
By leaving this comment I was hoping that a reviewer would be able to enlighten me.

Reply:
I am not sure I follow; `mark_as_trap_handler`
does not create a new block, so I am not sure
how it would interfere. `emit_tail_trywith` used
to call `Cmm.new_label`; what would be the
equivalent in the new version?

Reply:
The call to `Cmm.new_label ()` should still be there, when computing `handlers_map`

Reply:
Ah, yes, sorry. And I was misrimembering
what `mark_as_trap_handler` was supposed
to do. While it does not create a new block,
it changes the label. In the new implementation,
shouldn't `mark_as_trap_handler` only set
`is_trap_handler` to `true`?

Reply:
That looks correct, thanks. I'll implement that.